Python/specialize.c

```diff
@@ -945,6 +945,10 @@ specialize_class_load_attr(PyObject *owner, _Py_CODEUNIT *instr,
                              PyObject *name)
 {
     _PyLoadMethodCache *cache = (_PyLoadMethodCache *)(instr + 1);
+    if (!PyType_CheckExact(owner) || _PyType_Lookup(Py_TYPE(owner), name)) {
```

Comment:
You could safely generalize this to any immutable metaclass.
`Py_TYPE(owner)->tp_flags & Py_TPFLAGS_IMMUTABLETYPE`

Reply:
Just to clarify, you're only referring to the `PyType_CheckExact` call, right? The `_PyType_Lookup` call stays?

Wouldn't we also need to add something like `Py_TYPE(owner)->tp_getattro == PyType_Type.tp_getattro` too to protect against immutable types that define `__getattribute__`?

Reply:
> Just to clarify, you're only referring to the PyType_CheckExact call, right?

Yes

> Wouldn't we also need to add something like Py_TYPE(owner)->tp_getattro == PyType_Type.tp_getattro too

And probably some other special case I've forgotten.
Let's just stick with the `PyType_CheckExact(owner)` to be on the safe side.